### ISSUE TITLE
V2_MetaFunction using script:ld+json twice results in same key warning

### ISSUE THREAD

**Issue by wilcoschoneveld** [remix-run/remix#6953]
### What version of Remix are you using?

1.18.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```
export const meta: V2_MetaFunction = () => {
  return [
    {
      "script:ld+json": {
        "@context": "https://schema.org",
        "@type": "Product",
        ...
      },
    },
    {
      "script:ld+json": {
        "@context": "https://schema.org",
        "@type": "FAQPage",
        ...
      },
    },
  ];
};
```

### Expected Behavior

Should add two `<script type="application/ld+json">` tags without warning

### Actual Behavior

Adds two `<script type="application/ld+json">` tags with warning:

![image](https://github.com/remix-run/remix/assets/3064588/51020794-be5b-448c-86bf-7c43611a5ef6)

Culprit:
https://github.com/remix-run/remix/blob/97d82db0556ba8b40518898a75d6838e0bf3ced1/packages/remix-react/components.tsx#L834C36-L834C36

**Comment by kiliman** [comment]
Yes, it looks like the `key` should include the JSON-encoded value
```ts
<script 
  key={`script:ld+json:${json}`}
  type="application/ld+json"
  dangerouslySetInnerHTML={{ __html: json }}
/>
```

**Comment by brophdawg11** [comment]
This is resolved by #6954 and should be available once v2 is released.
